packages/web-components/@groupby/elements-products/test/unit/products.test.ts

```diff
@@ -67,26 +97,66 @@ describe('Products Sayt Component', () => {
   itShouldExtendClass(() => component, ProductsBase);
 
   describe('connectedCallback', () => {
+    let cacheResponseEventName;
+
+    beforeEach(() => {
+      cacheResponseEventName = component.cacheResponseEventName = 'sayt-cache-response';
+    });
+
     itShouldCallParentMethod(() => component, 'connectedCallback');
 
-    it('should set up an event listener for a sayt products event to set products', () => {
-      const addEventListener = sinon.stub(window, 'addEventListener');
+    it('should add event listeners for sayt products events', () => {
+      const addEventListener = stub(window, 'addEventListener');
```

Comment:
This function will always add event listeners, so we should stub this in `beforeEach` so we don't leak listeners.

Reply:
@jmbtutor it seems to me that the addListeners and RemoveListeners are set at the correct scope/place within products-sayt and products-search tests. They are not currently used in multiple tests within the connected and disconnected callbacks and so were set within the one test that uses them respectively. That seems correct to me but are you suggesting that addListener should be set in a beforeEach for the connectedCallback even though it is not used in multiple tests within the connectedCallback section? Or some other higher level beforeEach? Sorry Im just wanting to make sure I understand your feedback, and I'm unsure if setting these listeners when they arent used is correct? Thanks!

Reply:
explained and done